### PR TITLE
🎟️ [OJS] Payment request coupon fixes

### PR DESCRIPTION
**Pull request by syvlabs**

### CODE REVIEW

File: lib/components/_common/frame.tsx [hunk 1]
```diff
@@ -56,7 +56,9 @@ const ElementFrame: FC<ElementFrameProps> = (props) => {
 
   return (
     <iframe
+      id={`ojs-${subPath}-element`}
       name={`${subPath}-element`}
+      className="ojs-iframe"
```
Comment (syvlabs) [inline lib/components/_common/frame.tsx:61]: Not using these yet but I think it's nice to have them

File: lib/hooks/use-payment-requests.ts [hunk 2]
```diff
@@ -113,3 +104,79 @@ export const usePaymentRequests = (
...
+  const testerPR = await createStripePaymentRequest(stripePubKey, DUMMY_AMOUNT_ATOM, 'usd');
+  const canMakePayment = await testerPR.canMakePayment();
+  testerPR.abort();
+  if (!canMakePayment) {
+    throw new Error(`Cannot make payment with ${provider} for this session`);
+  }
+  return canMakePayment[OUR_PROVIDER_TO_STRIPES[provider]];
+};
+
+const startPaymentRequestUserFlow = async (
```
Comment (syvlabs) [inline lib/hooks/use-payment-requests.ts:140]: Refactored this out of the big hook above

File: lib/utils/cde-connection.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  | {
+      status: 'waiting';
+      conn: Connection<CallSender>;
+    }
+  | {
+      status: 'connected';
+      conn: CdeConnection;
+    };
+
+export const useCDEConnection = (): HookReturnType => {
```
Comment (syvlabs) [inline lib/utils/cde-connection.ts:34]: Core of the connection @zeyarpaing

File: lib/utils/cde_models.ts [hunk 4]
```diff
@@ -0,0 +1,218 @@
+import { z } from 'zod';
```
Comment (syvlabs) [inline lib/utils/cde_models.ts:1]: Just copy pasted from CDE

File: lib/components/form.tsx [hunk 5]
```diff
@@ -352,9 +361,13 @@ const ElementsForm: FC<ElementsFormProps> = (props) => {
       if (existingIframe) return;
 
       console.log('[form] Registering iframe:', iframe);
+      if (iframes.length === 0) {
+        // Only do the first iframe
+        connectToCdeIframe(iframe);
```
Comment (zeyarpaing) [inline lib/components/form.tsx:366]: @syvlabs , could you please explain more about this? will this also work on separate elements?

**Review by zeyarpaing — APPROVED** [review]

**Pull request merged**